hypervisor/src/hypervisor/syscall.rs

```diff
@@ -0,0 +1,168 @@
+//! Implementation of the mechanism to perform system calls
+
+use std::io::IoSliceMut;
+use std::os::fd::RawFd;
+use std::time::{Duration, Instant};
+
+use a653rs_linux_core::mfd::Mfd;
+use a653rs_linux_core::syscall::{SyscallRequ, SyscallResp};
+use anyhow::{bail, Result};
+use nix::sys::socket::{recvmsg, ControlMessageOwned, MsgFlags};
+use nix::{cmsg_space, unistd};
+use polling::{Event, Poller};
+
+/// Receives an FD triple from fd
+fn recv_fd_triple(fd: RawFd) -> Result<[RawFd; 3]> {
+    let mut cmsg = cmsg_space!([RawFd; 3]);
+    let mut iobuf = [0u8];
+    let mut iov = [IoSliceMut::new(&mut iobuf)];
+    let res = recvmsg::<()>(fd, &mut iov, Some(&mut cmsg), MsgFlags::empty())?;
```

Comment:
we may need the `MSG_DONTWAIT` Flag here

Reply:
Not really. This function only gets called after the poller decided that there is data readable on that file descriptor. Because we are dealing with Unix sockets here, this state, unlikely with inet sockets where a client could disconnect in the mean-time, cannot change to my knowledge.